docs/proposals/cluster-discovery-api.md

```diff
@@ -0,0 +1,91 @@
+# Cluster Discovery API
+
+Author: [@mikedanese](https://github.com/mikedanese)
+
+#### Abstract
+
+This document describes a mechanism for a kubelet to discover, from a trusted source, the
```

Comment:
From last sig-cluster-lifecycle meeting it sounded like this would also be used for discovery of masters/API servers. Where would etcd fit into this? I assume we don't really want to duplicate the etcd discovery service but having a discovery service be able to connect all components of the cluster would be really nice.